File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Integer i = 0;        
        HashMap<Integer,Integer> map = new HashMap<>(); 
 
        while(i < nums.length) {
            if(map.containsKey(nums[i]))
               return new int[]{map.get(nums[i]).intValue(),i.intValue()};  

           Integer difference = target-nums[i];
        
           map.put(difference,i);
          i++;
    }

    return new int[]{};
    }
}

File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int):
        mapp = dict()
        
        for i in range(len(nums)):
            if(target - nums[i] in mapp):
                return [i, mapp[target-nums[i]]]
            else:
                mapp[nums[i]] = i
            
        
        